### PR TITLE
Add `make_latest` to `GitRelease.update_release`

### PR DESCRIPTION
**Pull request by fb929**
1. The problem:
def "update_release" can not make release as "latest"

2. How solving the problem:
adding optional parameter "make_latest", from original api doc: https://docs.github.com/en/free-pro-team@latest/rest/releases/releases?apiVersion=2022-11-28#update-a-release--code-samples

### CODE REVIEW

**Comment by trim21** [comment]
can you also update document link in docstring? current link is out of data.

**Comment by pfarrell** [comment]
@fb929 I was just about to submit a PR to do exactly this.  
Are you planning to complete this PR?  This is a very feature I needed and had to implement by calling the github rest api directly.  I'd much prefer to have support in PyGithub.

File: github/GitRelease.py [hunk 1]
```diff
@@ -188,6 +189,9 @@ def update_release(
             "draft": draft,
             "prerelease": prerelease,
         }
+        if make_latest is not NotSet:
+            assert isinstance(make_latest, bool), make_latest
```
Comment (EnricoMi) [inline github/GitRelease.py:193]: documentation says, this is a string
```suggestion
            assert make_latest in ["true", "false", "legacy"], make_latest
```

**Review by EnricoMi — CHANGES_REQUESTED** [review]
While you are here, can you please add `discussion_category_name` as well?

**Comment by EnricoMi** [comment]
@pfarrell feel free to fork @fb929's repo and branch and create a new PR to finish this. This way, @fb929's contribution is credited and we get this over the finish line.

**Comment by EnricoMi** [comment]
Looks like `message` has been renamed to `body`. This should be reflected in another PR to avoid this to break in the future. https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28

File: github/GitRelease.py [hunk 2]
```diff
@@ -165,6 +165,7 @@ def update_release(
         message: str,
         draft: bool = False,
         prerelease: bool = False,
+        make_latest: Opt[bool] = NotSet,
```
Comment (EnricoMi) [inline github/GitRelease.py:168]: Adding this here may break user code, can you add this at the end of the argument list?

**Comment by EnricoMi** [comment]
@pfarrell @fb929 can we get this over the finish line?

**Comment by treee111** [comment]
I tackled the PR review of this PR in the linked PR. @trim21 and @EnricoMi feel free to review.

> Looks like `message` has been renamed to `body`. This should be reflected in another PR to prevent this from breaking in the future. https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28

we already send/post that as `body` to the Github API and have `message` only as input to PyGithub if I am correct? So no need for adjustment (because that would break user code). See:
https://github.com/PyGithub/PyGithub/blob/7e7653fb7cb054d3b03dd5f89dea1a2a79350827/github/GitRelease.py#L204-L210

**Comment by pfarrell** [comment]
> @pfarrell @fb929 can we get this over the finish line?

@EnricoMi, apologies for missing the alerts on this, thanks @@treee111 for picking this up

**Comment by treee111** [comment]
You are welcome @pfarrell. Feel free to test the coding and comment if it worked for you! Helps the PR getting merged :-)

**Comment by EnricoMi** [comment]
Closed in favour of #2888.

**Pull request closed without merging**